f/medical/fn_famAddHealAction.sqf

```diff
@@ -62,26 +60,23 @@ private _healCodeStart = {
 }; 
 
 // Progress Code
-private _healCodeProg = { 
-	params ["_target", "_caller", "_actionId", "_arguments"]; 
-}; 
+private _healCodeProg = {}; 
 
 // Completed Code
 private _healCodeComp = { 
-	params ["_target", "_caller", "_actionId", "_arguments"]; 
-
 	// this is needed to protect against BI bugs that remove all actions.
 	_caller setVariable ["f_var_fam_flag",false];
 
 	// Medic heals to full only if they have a medikit. TODO CLS Support?
-	if (_caller getUnitTrait 'Medic' && 'Medikit' in items _caller) then {
+	if (_caller getUnitTrait 'Medic' && (_caller call f_fnc_famHasFAK >= 1)) then {
 		_target setDamage 0;
 		hint "Patient healed fully with Medikit"; // feedback on resource consumption.
 	} else {	
 		_target setDamage 0.25;
-		if ("FirstAidKit" in items _caller) then {
-			_caller removeItem "FirstAidKit"; 
-			hint format ["Patient healed partially with FAK, %1 remaining. Medic required for further healing.",count (items _caller select {_x == "FirstAidKit"})]; // feedback on resource consumption.
+		if (_caller call f_fnc_famHasFAK == 0) then {
+			private _FAKs = (items _caller select {(getNumber (configFile >> "CfgWeapons" >> _x >> "ItemInfo" >> "type")) == 401});
```

Comment:
Would it be worth pulling this into a function as well since it's repeated in a few places? E.g. `famListFAKs`

Reply:
@costno